source/credentials_provider_process.c

```diff
@@ -0,0 +1,374 @@
+    if (!credentials) {
+        AWS_LOGF_INFO(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "(id=%p) Process credentials provider failed to parse credentials from command output (output is not "
+            "logged in case sensitive information).",
+            (void *)provider);
+        goto on_finish;
+    }
+
+    AWS_LOGF_INFO(
+        AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+        "(id=%p) Process credentials provider successfully sourced credentials.",
+        (void *)provider);
+    ret = AWS_OP_SUCCESS;
+
+on_finish:
+    callback(credentials, user_data);
+    aws_run_command_result_cleanup(&result);
+    aws_credentials_destroy(credentials);
+    return ret;
+}
+
+static void s_credentials_provider_process_destroy(struct aws_credentials_provider *provider) {
+    struct aws_credentials_provider_process_impl *impl = provider->impl;
+    if (impl) {
+        aws_string_destroy_secure(impl->command);
+    }
+    aws_credentials_provider_invoke_shutdown_callback(provider);
+    aws_mem_release(provider->allocator, provider);
+}
+
+AWS_STATIC_STRING_FROM_LITERAL(s_credentials_process, "credential_process");
+static struct aws_byte_cursor s_default_profile_name_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("default");
+
+static struct aws_profile_collection *s_load_profile(struct aws_allocator *allocator) {
+
+    struct aws_profile_collection *config_profiles = NULL;
+    struct aws_string *config_file_path = NULL;
+
+    config_file_path = aws_get_config_file_path(allocator, NULL);
+    if (!config_file_path) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to resolve config file path during process credentials provider initialization: %s",
+            aws_error_str(aws_last_error()));
+        goto on_error;
+    }
+
+    config_profiles = aws_profile_collection_new_from_file(allocator, config_file_path, AWS_PST_CONFIG);
+    if (config_profiles != NULL) {
+        AWS_LOGF_DEBUG(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Successfully built config profile collection from file at (%s)",
+            aws_string_c_str(config_file_path));
+    } else {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to build config profile collection from file at (%s) : %s",
+            aws_string_c_str(config_file_path),
+            aws_error_str(aws_last_error()));
+        goto on_error;
+    }
+
+    aws_string_destroy(config_file_path);
+    return config_profiles;
+
+on_error:
+    aws_string_destroy(config_file_path);
+    aws_profile_collection_destroy(config_profiles);
+    return NULL;
+}
+
+static void s_check_or_get_with_profile_config(
+    struct aws_allocator *allocator,
+    struct aws_profile *profile,
+    struct aws_string **target,
+    const struct aws_string *config_key) {
+
+    if (!allocator || !profile || !config_key) {
```

Comment:
I don't think there's a need to allocate anything here or clean up an existing value.

Reply:
If *target is not null but the length is zero, I think still need to destroy and allocate new one. change it to byte_buf would be easier. The part of byte_buf I don't like is the complexity to process trimmed data.